### PR TITLE
Add example: live FFT

### PR DESCRIPTION
**Pull request by magnunor**
### Description of the change

- This pull request adds an example showing how to get a live fourier transform from an interactive region of interest.

### Progress of the PR
- [x] Change implemented (can be split into several points),
- [x] add an changelog entry in the `upcoming_changes` folder (see [`upcoming_changes/README.rst`](https://github.com/hyperspy/hyperspy/blob/RELEASE_next_minor/upcoming_changes/README.rst)),
- [ ] Check formatting changelog entry in the `readthedocs` doc build of this PR (link in github checks)
- [ ] ready for review.

### Discussion

The FFT of the example signal (`hs.data.atomic_resolution_image()`), looks really strange. So I'm tempted to also add a new example atomic resolution signal.

### CODE REVIEW

File: examples/region_of_interest/fast_fourier_transform_live.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+#%%
+# Create the roi, here a :py:class:`~.api.roi.RectangularROI`:
+roi = hs.roi.RectangularROI()
+
+#%%
+# Slice signal with roi with the ROI. By using the `interactive` function, the
+# output signal ``sliced_signal`` will update automatically.
+# The ROI will be added automatically on the signal figure.
+s.plot()
+sliced_signal = roi.interactive(s)
```
Comment (ericpre) [inline examples/region_of_interest/fast_fourier_transform_live.py]:
```suggestion
sliced_signal = roi.interactive(s, recompute_out_event=None)
```

File: examples/region_of_interest/fast_fourier_transform_live.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+# Choose the second figure as gallery thumbnail:
+# sphinx_gallery_thumbnail_number = 2
+
+#%%
+# Get the FFT of this sliced signal, and plot it
+s_fft = hs.interactive(sliced_signal.fft, shift=True)
```
Comment (ericpre) [inline examples/region_of_interest/fast_fourier_transform_live.py]:
```suggestion
s_fft = hs.interactive(sliced_signal.fft, apodization=True, shift=True, recompute_out_event=None)
```

**Review by ericpre — COMMENTED** [review]
I suggested some improvement.

It would be good to improve the `hs.data.atomic_resolution_image` to make a more realistic (noise, width of atomic columns, larger images, etc.), something similar as https://github.com/ericpre/gpa/blob/main/gpa/datasets/utils.py, but this can be done in a separate PR.

**Comment by magnunor** [comment]
I agree with improving the `hs.data.atomic_resolution_image` function. I might do that after finishing this PR.

I added your suggestions, and an "upcoming_changes`. I'm not sure why the RTD and documentation building test is failing. Seems to be related to the towncrier:

```
Extension error (sphinxcontrib.towncrier.ext):
Handler <bound method TowncrierDraftEntriesEnvironmentCollector.get_outdated_docs of <sphinxcontrib.towncrier.ext.TowncrierDraftEntriesEnvironmentCollector object at 0x7f5d89531250>> for event 'env-get-outdated' threw an exception (exception: find_fragments() takes 3 positional arguments but 4 were given)
```

**Comment by ericpre** [comment]
Most likely related to the latest release of towncrier: https://github.com/sphinx-contrib/sphinxcontrib-towncrier/issues/92.

The commit [5e2b2df](https://github.com/hyperspy/hyperspy/pull/3400/commits/5e2b2df1a017a2643469196596cc4ef0f72571b0) shouldn't be in this PR, is it worth trying to remove it?

**Comment by magnunor** [comment]
> The commit [5e2b2df](https://github.com/hyperspy/hyperspy/pull/3400/commits/5e2b2df1a017a2643469196596cc4ef0f72571b0) shouldn't be in this PR, is it worth trying to remove it?

This commit has been removed.

**Review by ericpre — APPROVED** [review]

**Pull request merged**